xprocess.py

```diff
@@ -59,7 +59,7 @@ def isrunning(self):
             proc = psutil.Process(self.pid)
         except psutil.NoSuchProcess:
             return False
-        return proc.is_running()
+        return proc.is_running() and proc.status() != psutil.STATUS_ZOMBIE
```

Comment:
Hm, ignoring zombie processes might not be desired for all use cases. I, for one, have had situations where I needed to execute a certain group of assertions only after a process had been **completely** (no zombies) terminated. We could try something like the following to make your changes an optional behaviour.

```python
def isrunning(self, **kwargs):
    ignore_zombies = kwargs.pop("ignore_zombies", False)
    if kwargs:
        raise TypeError("unknown keyword arguments: {}".format(kwargs.keys()))
    if self.pid is None:
        return False
    try:
        proc = psutil.Process(self.pid)
    except psutil.NoSuchProcess:
        return False
    return proc.is_running() and (not ignore_zombies or proc.status() != psutil.STATUS_ZOMBIE) 
```

Reply:
Ok, sure, it sounds reasonable to make it configurable if there are cases when both behaviors are wanted. Just out of curiosity, when does it make sense to not ignore zombie processes? I'm thinking that those perhaps should be ignored by default, unless you're not that happy about changing the behavior.

Reply:
The zombies themselves are not what I’m worried about since they take virtually no resources other than proc IDs. Considering all children processes first become zombies before being reaped, the ones that stay as zombies for extended periods of time might hint to a bug in the parent process. Hiding such potential bugs by ignoring zombies is what worries me. Considering that external "dependencies" handled by xprocess might actually be the objects being tested (e.g. someone has developed their own socket server and wants to test how it performs using a simple client application), unless there’s a real benefit in zombies **always** being ignored or any downside in making this behaviour optional that I am not seeing here, I believe something along the lines of the suggested changes would be a beneficial simple adjustment to your (already great) contribution.

Reply:
Ok, thanks. And that's primarily a case when expecting the external process to live as long as a single pytest run, but not longer, right?

If I'm not mistaken, a short-lived process will always become a zombie process until the pytest run is completed, but should never become one otherwise. Do you think it would be more or less confusing for the user to  ignore zombie processes by default for processes that has been started in this run? I think that would just work, but I might very well be missing use cases.

Reply:
`If` I'm not mistaken, a short-lived process will always become a zombie...

A **child** process will always become a zombie before having its status collected by the parent. How long a process "lasts" has no relation to it becoming a zombie or not as far as I can tell. 

The code sample in my suggestion was intended to be just a starting point for the idea I proposed, which is making it configurable. If you would like to ignore zombies by default, I can't think of any potential problems in doing that at the moment. So, as long as we still give the user a way of optionally (i.e. using `ignore_zumbies=False`) detecting such processes (for previously mentioned reasons), feel free to do it.

By the way, since a new release is out,  please rebase this to solve conflicts. It would be great if you could also add a changelog and a docstring to `XProcessInfo.isrunning` describing the new behaviour you are implementing. Again, thanks for your contribution!

Reply:
> A **child** process will always become a zombie before having its status collected by the parent. How long a process "lasts" has no relation to it becoming a zombie or not as far as I can tell.

Ok, you're probably right, but my understanding of https://github.com/giampaolo/psutil/issues/1008 was that all processes that excited before pytest became zombie because of how psutil works. Maybe that's not correct?

Reply:
And, for the record, I'll get to the changes you're suggesting as soon as I find the time. Thanks for a constructive discussion!

Reply:
@northernSage, I believe that I have fixed your comments now.

Reply:
...and now CI is green. 💚